### PR TITLE
feat: Projects CLI, add sort-by fields

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Work in progress on a handful of workspaces / projects issues:
- adding archive/unarchive to projects cli, and moving experiments between projects
- removing a.sort to fix a bug in GetProjectExperiments
- add two sortable fields (forked-from and resource-pool); Due to a weird bug where GetExperiments and GetProjectExperiments were too similar for the Python bindings, I added a third field
- make experiment labels filter-able by project_id

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/cli/experiment.py [hunk 1]
```diff
@@ -747,6 +748,18 @@ def unarchive(args: Namespace) -> None:
     print("Unarchived experiment {}".format(args.experiment_id))
 
 
+@authentication.required
+def move_project(args: Namespace) -> None:
```
Comment (EmilyBonar) [inline harness/determined/cli/experiment.py:752]: question: Is this meant to be called `move_project`? The code seems to be implementing move_experiment.

File: master/internal/api_workspace.go [hunk 2]
```diff
@@ -85,8 +85,6 @@ func (a *apiServer) GetWorkspaceProjects(ctx context.Context,
 		return nil, err
 	}
 
-	a.sort(resp.Projects, req.OrderBy, req.SortBy,
```
Comment (EmilyBonar) [inline master/internal/api_workspace.go:88]: question: Does this still work even with this `a.sort` removed?

**Review by EmilyBonar — APPROVED** [review]
LGTM

File: harness/determined/cli/experiment.py (hunk 1 above)
Comment (mapmeld) [reply harness/determined/cli/experiment.py]: ✅  fixed

File: master/internal/api_workspace.go (hunk 2 above)
Comment (mapmeld) [reply master/internal/api_workspace.go:88]: My new understanding is that `a.sort` is for objects which aren't sorted by the database for some reason (for example, it's used on checkpoints and in notebooks, but never on GetExperiments).
e2e_tests include sorting by id and name to verify orders that workspaces and projects are returned

File: webui/react/src/services/apiConfig.ts [hunk 3]
```diff
@@ -400,11 +400,11 @@ export const getExpTrials: Service.DetApi<
 };
 
 export const getExperimentLabels: Service.DetApi<
-  Service.EmptyParams, Api.V1GetExperimentLabelsResponse, string[]
+  Service.ExperimentLabelsParams, Api.V1GetExperimentLabelsResponse, string[]
 > = {
   name: 'getExperimentLabels',
   postProcess: (response) => response.labels || [],
-  request: (options) => detApi.Experiments.getExperimentLabels(options),
+  request: (options) => detApi.Experiments.getExperimentLabels(options.project_id || 0),
```
Comment (EmilyBonar) [inline webui/react/src/services/apiConfig.ts:407]: I believe this should be `request: (params, options) => detApi.Experiments.getExperimentLabels(params.project_id, options),`

**Pull request merged**